### PR TITLE
Add our IGRF

### PR DESCRIPTION
**Pull request by thetazero**
Cleaned up version of fifth order IGRF model from Kevin Tracy.

### CODE REVIEW

File: state_machine/applications/flight/lib/IGRF.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+except ImportError:
+    import numpy as np
+import math
+
+def reset_array(input_array):
+    for i in range(len(input_array)):
+        input_array[i] = 0.0
+
+def unix_time_to_years_since_2020(unix_time):
+    twentytwenty = 1577854800  # 2022-01-01 00:00:00 time stamp
```
Comment (yashika-batra) [inline state_machine/applications/flight/lib/IGRF.py:16]: it seems like you want this to be the first date of the year 2020 -- i'm not sure if the value is off or just the comment, but you assign the timestamp for 1/1/2022 to something that's meant to be 1/1/2020

File: state_machine/applications/flight/lib/IGRF.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    t = (unix_time - twentytwenty) / 31557600  # seconds in a Julian astronomical year
+    return t
+
+def igrf13_5(gh, date, latitude_degrees, elongitude_degrees, r_norm_km, cl, sl, p, q):
```
Comment (yashika-batra) [inline state_machine/applications/flight/lib/IGRF.py:20]: i would add a comment somewhere that tells us what r_norm_km, cl, sl, p, and q stand for

File: state_machine/applications/flight/lib/IGRF.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    # reset the lists that are passed by reference
+    reset_array(cl)
+    reset_array(sl)
+    reset_array(p)
+    reset_array(q)
+
+    # colatitude
+    colat = 90 - latitude_degrees
+
+    # Declaration of variables
```
Comment (yashika-batra) [inline state_machine/applications/flight/lib/IGRF.py:31]: Same thing here: maybe add inline comments for each of these explaining what they stand for, or (this is a future solution) add a comment that links users to the documentation that explains this

File: state_machine/applications/flight/lib/IGRF.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+    ll = 0
+
+    # nc = int(nmx * (nmx + 2))
+    nc = 35
+
+    # kmx = int((nmx + 1) * (nmx + 2) / 2)
+    kmx = 21
+
+    r = r_norm_km
```
Comment (yashika-batra) [inline state_machine/applications/flight/lib/IGRF.py:54]: Same thing here about comments/clarity

**Review by yashika-batra — CHANGES_REQUESTED** [review]

**Comment by yashika-batra** [comment]
Two major changes: 
1. I think you added the date for 2022 instead of 2020 in your unix_time_to_years_since_2020 function
2. There are a lot of 1-2 letter variable names, which reduces the clarity of the code. I would suggest to do one of the 3 things:
- rename variables
- add inline comments explaining what each variable stands for
- (this is a future solution) add a comment that links users to the documentation that explains what each variable stands for

Apart from that it looks great!

File: state_machine/applications/flight/lib/IGRF.py (hunk 1 above)
Comment (thetazero) [reply state_machine/applications/flight/lib/IGRF.py:16]: Thanks for pointing this out!

File: state_machine/applications/flight/lib/IGRF.py (hunk 2 above)
Comment (thetazero) [reply state_machine/applications/flight/lib/IGRF.py:20]: honestly igrf13_5 is just copy pasted from the port and modified to use unix time. I think it's best to just leave the internals alone even though their a mess. I think the interface functions are pretty well documented. 

I've added an _ to the function name to indicate that it's internal and not meant to be used.

**Comment by thetazero** [comment]
Thanks for the review @yashika-batra. 
I'm not going to refactor the internal IGRF13 code because even though it's a total mess, it's just a copy pasted library. 

I'm trying to make sure that interface this library provides is easy to use though.

**Comment by yashika-batra** [comment]
> Thanks for the review @yashika-batra. I'm not going to refactor the internal IGRF13 code because even though it's a total mess, it's just a copy pasted library. I don't think it's really worth our time to try and clear it up.
> 
> I'm trying to make sure that interface this library provides is easy to use though.

Got it! I didn't realize it was a copy pasted library. In that case, this is good to go! I'll approve the merge.

**Review by yashika-batra — APPROVED** [review]

**Pull request merged**